src/common/socket.c

```diff
@@ -238,7 +238,11 @@ fd_set readfds;
 
 // Maximum packet size in bytes, which the client is able to handle.
 // Larger packets cause a buffer overflow and stack corruption.
-static size_t socket_max_client_packet = 24576;
+#if PACKETVER >= 20131223
+static size_t socket_max_client_packet = 0x10000;
```

Comment:
But there is a difference between `if (len < 0x10000) { /* accept the packet */ }` and `if (len > 0x10000) { /* reject the packet */ }`. Both have a 0x10000, but accept different ranges. 0x10000, in our case, would mean a buffer of 0x10001.